util/include/lru_cache.hpp

```diff
@@ -27,21 +27,12 @@ class LruCache {
   LruCache(size_t capacity) : capacity_(capacity) { map_.reserve(capacity); }
 
   void put(const Key& key, const Value& value) {
-    stats_.puts++;
-    auto iter = map_.find(key);
-    if (iter != map_.end()) {
-      moveToFront(iter->second.second);
-      iter->second.first = value;
-    } else {
-      if (keys_.size() == capacity_) {
-        map_.erase(keys_.back());
-        keys_.pop_back();
-      }
-      keys_.push_front(key);
-      map_.insert({key, {value, keys_.begin()}});
-    }
+    putImp(std::forward<const Key&>(key), std::forward<const Value&>(value));
   }
 
+  // using this rvalue put will invalidate any usage of the key-value that were passed to the method.
+  void put(Key&& key, Value&& value) { putImp(std::forward<Key&&>(key), std::forward<Value&&>(value)); }
```

Comment:
I think we don't need to do `std::forward` here. Instead, we need to do it inside `putImp()`, instead of the `std::moves` there.